src/segments/umbraco.go

```diff
@@ -0,0 +1,231 @@
+package segments
+
+import (
+	"encoding/xml"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Umbraco struct {
+	props properties.Properties
+	env   platform.Environment
+
+	FoundUmbraco    bool
+	IsModernUmbraco bool
+	IsLegacyUmbraco bool
+	Version         string
+}
+
+const (
+	umbracoFolderName = "umbraco"
+	umbracoWebConfig  = "web.config"
+)
+
+type CSProj struct {
+	PackageReferences []struct {
+		Name    string `xml:"Include,attr"`
+		Version string `xml:"Version,attr"`
+	} `xml:"ItemGroup>PackageReference"`
+}
+
+type WebConfig struct {
+	AppSettings []struct {
+		Key   string `xml:"key,attr"` // TODO: What happens if the web.config has the attribute as uppercase Key="" ?
+		Value string `xml:"value,attr"`
+	} `xml:"appSettings>add"`
+}
+
+type FindUmbracoResult struct {
+	FoundUmbracoFolder bool
+	FoundWebConfig     bool
+	FoundCSProj        bool
+	FilePath           string
+}
+
+func (u *Umbraco) Enabled() bool {
+	u.env.Debug("UMBRACO: Checking if we enable segment")
+
+	// Checks if the current folder contains an umbraco folder
+	// If not it then checks each parent until the root
+	findUmbracoResults, err := u.TryFindUmbracoInParentDirsOrSelf()
+
+	if err != nil {
+		u.env.Debug("UMBRACO: Error while searching for Umbraco folder and files")
+		u.env.Debug(err.Error())
+		return false
+	}
+
+	if !findUmbracoResults.FoundUmbracoFolder {
+		u.env.Debug("UMBRACO: No Umbraco folder found")
+		u.FoundUmbraco = false
+		return false
+	}
+
+	// We have found an Umbraco folder between CWD and root
+	// Along with finding a web.config file at the same level as the Umbraco folder
+	// OR one or more .csproj files at the same level as the Umbraco folder
+
+	// Modern .NET Core based Umbraco
+	if findUmbracoResults.FoundCSProj {
+		u.env.Debug("UMBRACO: Checking for modern Umbraco as we have found .csproj file")
+		return u.TryFindModernUmbraco(findUmbracoResults.FilePath)
+	}
+
+	// Legacy .NET Framework based Umbraco
+	if findUmbracoResults.FoundWebConfig {
+		u.env.Debug("UMBRACO: Checking for legacy Umbraco as we have found web.config file")
```

Comment:
You already log this in TryFindUmbracoInParentDirsOrSelf() so it's a bit double.